java-shared-dependencies/unmanaged-dependency-check/src/main/java/com/google/cloud/UnmanagedDependencyCheck.java

```diff
@@ -0,0 +1,78 @@
+package com.google.cloud;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.cloud.tools.opensource.classpath.ClassPathBuilder;
+import com.google.cloud.tools.opensource.classpath.DependencyMediation;
+import com.google.cloud.tools.opensource.dependencies.Bom;
+import com.google.cloud.tools.opensource.dependencies.MavenRepositoryException;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.eclipse.aether.resolution.ArtifactDescriptorException;
+import org.eclipse.aether.version.InvalidVersionSpecificationException;
+
+public class UnmanagedDependencyCheck {
+
+  public static void main(String[] args)
+      throws MavenRepositoryException, ArtifactDescriptorException, InvalidVersionSpecificationException {
+    checkArgument(args.length == 2, "The length of the inputs should be 2");
+    System.out.println(getUnmanagedDependencies(args[0], args[1]));
```

Comment:
What are good error message for library developers?

Reply:
Idea. "This pull request seems to be adding a new third-party dependency com.h2.xxx:h2 among the artifacts listed in BOM com.google.cloud:bigtable-bom. Please see go/cloud-sdk-java-dependency-governance. "